realm/realm-library/src/main/java/io/realm/RealmQuery.java

```diff
@@ -1951,6 +1951,28 @@ public long count() {
         return this;
     }
 
+    /**
+     * Returns the {@link Realm} instance to which this query belongs.
+     * <p>
+     * Calling {@link Realm#close()} on the returned instance is discouraged as it is the same as
+     * calling it on the original Realm instance causing the Realm to fully close invalidating the
```

Comment:
> causing the Realm

which may cause the Realm ...

_same comment for the other impl (RealmList.java etc)_